syntax = "proto3" ;

service Fileservice {
    rpc ReplicateFile (FileData) returns (ack);
}

message FileData {
    string initialReplicaServer = 1;
    bytes bytearray = 2;
    string vClock = 3;
    repeated string shortest_path = 4;
    int32 currentpos = 5;
}

message ack {
    bool success =1;
    string message = 2;
}